piptools/scripts/sync.py

```diff
@@ -60,15 +68,15 @@ def cli(dry_run, force, find_links, index_url, extra_index_url, no_index, quiet,
     to_install, to_uninstall = sync.diff(requirements, installed_dists)
 
     install_flags = []
-    for link in find_links or []:
+    for link in repository.finder.find_links or []:
         install_flags.extend(['-f', link])
-    if no_index:
+    if not repository.finder.index_urls:
         install_flags.append('--no-index')
-    if index_url:
-        install_flags.extend(['-i', index_url])
-    if extra_index_url:
-        for extra_index in extra_index_url:
-            install_flags.extend(['--extra-index-url', extra_index])
+    for (i, index_url) in enumerate(repository.finder.index_urls):
```

Comment:
This is not quite the same intent. I see that you are reconstructing the parameter invocation based upon the index_urls, but that exposes too much of the implementation detail. I don't quite grasp why you are doing this.

Reply:
The point here is that the call to `pip.req.parse_requirements` above (line 56) will update the flags in the `repository.finder` (which is passed to it) according to the flags specified in the txt file. Therefore the `repository.finder.index_urls` will contain combination of command line options and txt file options and this is then used here to construct the options given to `pip install`.

Reply:
I don't understand what you mean by the "implementation detail", since there wasn't really any separation between implementation and interface here before my chagnes either. If implementation details should be hidden, then there should be a class like `RequirementsFileParser` which then could be used to parse the txt files and as a result would output the list of requirements and pip options from the parsed files. That would be nice improvement, but completely another issue.

Can you elaborate what do you mean by exposing the implementation detail here?